src/types/types.ts

```diff
@@ -793,6 +824,217 @@ export interface ServiceAccountInput {
   groups?: Groups;
 }
 
+export interface Model3DListRequest {
+  limit?: number;
+  /**
+   * Filter based on whether or not it has published revisions.
+   */
+  published?: boolean;
+}
+
+export interface CreateRevision3D {
+  /**
+   * True if the revision is marked as published.
+   */
+  published?: boolean;
+  /**
+   * Global rotation to be applied to the entire model.
+   * The rotation is expressed by Euler angles in radians and in XYZ order.
+   */
+  rotation?: [boolean, boolean, boolean];
+  camera?: RevisionCameraProperties;
+  /**
+   * The file id to a file uploaded to Cognite's Files API.
+   * Can only be set on revision creation, and can never be updated. _Only FBX files are supported_.
+   */
+  fileId: CogniteInternalId;
+}
+
+export type Tuple3<T> = [T, T, T];
+
+export interface UpdateRevision3D {
+  id: CogniteInternalId;
+  update?: {
+    /**
+     * True if the revision is marked as published.
+     */
+    published?: SetField<boolean>;
+    /**
+     * Global rotation to be applied to the entire model.
+     * The rotation is expressed by Euler angles in radians and in XYZ order.
+     */
+    rotation?: SetField<Tuple3<number>>;
+    /**
+     * Initial camera target.
+     */
+    camera?: SetField<RevisionCameraProperties>;
+  };
+}
+
+export interface Revision3DListRequest {
+  /**
+   * Limits the number of results to be returned.
+   * The maximum results returned by the server is 1000 even if the limit specified is larger.
+   */
+  limit?: number;
+  /**
+   * Filter based on whether or not it has published revisions.
+   */
+  published?: boolean;
+}
+
+export interface RevisionCameraProperties {
+  /**
+   * Initial camera target.
+   */
+  target?: [number, number, number];
+  /**
+   * Initial camera position.
+   */
+  position?: [number, number, number];
+}
+
+export interface Revision3D {
+  /**
+   * The ID of the revision.
+   */
+  id: CogniteInternalId;
+  /**
+   * The file id.
+   */
+  fileId: CogniteInternalId;
+  /**
+   * True if the revision is marked as published.
+   */
+  published: boolean;
+  /**
+   * Global rotation to be applied to the entire model.
+   * The rotation is expressed by Euler angles in radians and in XYZ order.
+   */
+  rotation?: [number, number, number];
+  camera?: RevisionCameraProperties;
+  /**
+   * The status of the revision.
+   */
+  // tslint:disable-next-line
```

Comment:
Why?

Reply:
```
Refactor this union type to have less than 3 elements.
```

Reply:
Ah strange... Could you specify the exact rule when disable the line? Like this: https://github.com/cognitedata/cognitesdk-js/pull/87/files#diff-1bc3db66f407dd6ed36987395b7c5632R74

Reply:
fixed it by making separate type